### ISSUE TITLE
Disable un-await-ed Promises in async functions

### ISSUE THREAD

**Issue by hfhchan** [eslint/eslint#10847]
**Please describe what the rule should do:**
Disable un-awaited-ed Promises in async functions.  All promise assignments must be await-ed or assigned to a variable with a certain suffix (e.g. xxxPromise, xxx_promise) or matching a regex.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Enforces code style
[ x ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**
```js
async function getNews() {
   /* do something */ 
};

async function main() {
  const news = getNews(); // Warning/Error by Eslint here
  news.forEach( /* do something */ ) // Otherwise, Runtime error would happen here
}
```

```js
async function main() {
  const news = Promise.resolve().then(/* do something */); // Warning/Error by Eslint here
  news.forEach( /* do something */ ) // Otherwise, Runtime error would happen here
}
```
**Correct case:**
```js
async function getNews() {
   /* do something */ 
};

async function main() {
  const newsPromise = getNews(); // No Warning by Eslint
  const news = await newsPromise; // No Warning by Eslint
  news.forEach( /* do something */ ); // Code runs
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**
Forgetting to include await is a common error when refactoring Promise-heavy code (especially multiple nested Promises) into async/await. 

**Why support a suffix option**:
Multiple async tasks may be kicked off in succession before reading them (await), so there should be a clear way to indicate assignment to a promise is intended.  Probably allowing a suffix or a regex pattern, similar to how unused-vars can be overridden.

For example a use case that would be supported by allowing a suffix:

```js
async function getNews() { /* do something */ }
async function getEvents() { /* do something */ } 

// fetching news and events happen in parallel
const newsPromise = getNews();
const eventsPromise = getEvents();

// reading of news and events happen sequentially
const obj = {
  news: await newsPromise,
  events: await eventsPromise
};
```

**Comment by kaicataldo** [comment]
I have the same concerns about this as I do with https://github.com/eslint/eslint/issues/10829#issuecomment-429713012. A type system like TypeScript seems like a much better solution to this problem.
